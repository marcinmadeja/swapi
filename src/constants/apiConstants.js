export const SWAPI_URL = 'https://swapi.co/api';
export const SWAPI_FILMS = '/films/?format=json';

export const RANDOM_USERS = 'https://randomuser.me/api';
