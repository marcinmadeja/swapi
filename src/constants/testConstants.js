export const users = [
  {
    gender: 'male',
    name: {
      title: 'mr',
      first: 'terrence',
      last: 'watts',
    },
    location: {
      street: '5796 rolling green rd',
      city: 'bowral',
      state: 'victoria',
      postcode: 7762,
      coordinates: {
        latitude: '-79.9438',
        longitude: '-160.0279',
      },
      timezone: {
        offset: '-7:00',
        description: 'Mountain Time (US & Canada)',
      },
    },
    email: 'terrence.watts@example.com',
    login: {
      uuid: 'ef30b2c2-7575-42d1-b9e5-38030d64b184',
      username: 'tinygoose275',
      password: 'makaveli',
      salt: 'EwEwiYhD',
      md5: '67600680d1f62c990021d597e96b264d',
      sha1: '174b563d3f42d98f88e7bcaad064e647ee726faa',
      sha256: 'fc93dd24693ad0650413b01980ccc494dd3921991a8ab41e45cadf8610aa046a',
    },
    dob: {
      date: '1972-09-08T12:37:28Z',
      age: 45,
    },
    registered: {
      date: '2003-03-25T20:55:50Z',
      age: 15,
    },
    phone: '00-8437-1526',
    cell: '0476-938-903',
    id: {
      name: 'TFN',
      value: '686111438',
    },
    picture: {
      large: 'https://randomuser.me/api/portraits/men/24.jpg',
      medium: 'https://randomuser.me/api/portraits/med/men/24.jpg',
      thumbnail: 'https://randomuser.me/api/portraits/thumb/men/24.jpg',
    },
    nat: 'AU',
  },
  {
    gender: 'male',
    name: {
      title: 'mr',
      first: 'paulo',
      last: 'seyfarth',
    },
    location: {
      street: 'talstra√üe 94',
      city: 'bogen',
      state: 'bayern',
      postcode: 74635,
      coordinates: {
        latitude: '18.2360',
        longitude: '72.7154',
      },
      timezone: {
        offset: '-3:00',
        description: 'Brazil, Buenos Aires, Georgetown',
      },
    },
    email: 'paulo.seyfarth@example.com',
    login: {
      uuid: 'af5ee15a-da22-4264-8c94-7cd57e21bf97',
      username: 'crazybutterfly740',
      password: 'colt45',
      salt: '7LDAWnJY',
      md5: 'e5bbaee291478ce51472ec82d31f97c6',
      sha1: 'ea8722d0fc4ba69dc43e36661468f5bda5199799',
      sha256: 'e31a953387293ab446bf702c20013a91f1c91793fe59fe425a6e6502d502ad50',
    },
    dob: {
      date: '1969-06-18T22:13:53Z',
      age: 49,
    },
    registered: {
      date: '2004-09-15T22:48:16Z',
      age: 13,
    },
    phone: '0477-2629824',
    cell: '0175-1061186',
    id: {
      name: '',
      value: null,
    },
    picture: {
      large: 'https://randomuser.me/api/portraits/men/98.jpg',
      medium: 'https://randomuser.me/api/portraits/med/men/98.jpg',
      thumbnail: 'https://randomuser.me/api/portraits/thumb/men/98.jpg',
    },
    nat: 'DE',
  },
  {
    gender: 'male',
    name: {
      title: 'mr',
      first: 'mathias',
      last: 'mortensen',
    },
    location: {
      street: '9165 fredensvej',
      city: 'pandrup',
      state: 'nordjylland',
      postcode: 39076,
      coordinates: {
        latitude: '-12.9436',
        longitude: '101.5652',
      },
      timezone: {
        offset: '+7:00',
        description: 'Bangkok, Hanoi, Jakarta',
      },
    },
    email: 'mathias.mortensen@example.com',
    login: {
      uuid: '39d341b7-2f7b-4b4a-bfd4-6921922da746',
      username: 'beautifulladybug715',
      password: 'carpente',
      salt: 'KpPfkc93',
      md5: '5f976528c474e2c1479464183d7800d9',
      sha1: '6d25ac15f23bf114e05cb940e7c61925b3274746',
      sha256: 'cf715b8dc50773efd203d755f2b0c268fd588e5b98204545ad1a71d30b66e4b2',
    },
    dob: {
      date: '1990-04-03T15:42:14Z',
      age: 28,
    },
    registered: {
      date: '2014-05-19T01:46:06Z',
      age: 4,
    },
    phone: '04223613',
    cell: '62185716',
    id: {
      name: 'CPR',
      value: '722632-7968',
    },
    picture: {
      large: 'https://randomuser.me/api/portraits/men/13.jpg',
      medium: 'https://randomuser.me/api/portraits/med/men/13.jpg',
      thumbnail: 'https://randomuser.me/api/portraits/thumb/men/13.jpg',
    },
    nat: 'DK',
  },
];

export const swapiPeople = [
  {
    name: 'Leia Organa',
    height: '150',
    mass: '49',
    hair_color: 'brown',
    skin_color: 'light',
    eye_color: 'brown',
    birth_year: '19BBY',
    gender: 'female',
    homeworld: 'https://swapi.co/api/planets/2/',
    films: [
      'https://swapi.co/api/films/2/',
      'https://swapi.co/api/films/6/',
      'https://swapi.co/api/films/3/',
      'https://swapi.co/api/films/1/',
      'https://swapi.co/api/films/7/',
    ],
    species: [
      'https://swapi.co/api/species/1/',
    ],
    vehicles: [
      'https://swapi.co/api/vehicles/30/',
    ],
    starships: [],
    created: '2014-12-10T15:20:09.791000Z',
    edited: '2014-12-20T21:17:50.315000Z',
    url: 'https://swapi.co/api/people/5/',
  },
  {
    name: 'Obi-Wan Kenobi',
    height: '182',
    mass: '77',
    hair_color: 'auburn, white',
    skin_color: 'fair',
    eye_color: 'blue-gray',
    birth_year: '57BBY',
    gender: 'male',
    homeworld: 'https://swapi.co/api/planets/20/',
    films: [
      'https://swapi.co/api/films/2/',
      'https://swapi.co/api/films/5/',
      'https://swapi.co/api/films/4/',
      'https://swapi.co/api/films/6/',
      'https://swapi.co/api/films/3/',
      'https://swapi.co/api/films/1/',
    ],
    species: [
      'https://swapi.co/api/species/1/',
    ],
    vehicles: [
      'https://swapi.co/api/vehicles/38/',
    ],
    starships: [
      'https://swapi.co/api/starships/48/',
      'https://swapi.co/api/starships/59/',
      'https://swapi.co/api/starships/64/',
      'https://swapi.co/api/starships/65/',
      'https://swapi.co/api/starships/74/',
    ],
    created: '2014-12-10T16:16:29.192000Z',
    edited: '2014-12-20T21:17:50.325000Z',
    url: 'https://swapi.co/api/people/10/',
  },
];

