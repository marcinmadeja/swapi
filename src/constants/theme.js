import { createMuiTheme } from 'material-ui/styles';
import blueGrey from 'material-ui/colors/blueGrey';
import yellow from 'material-ui/colors/yellow';

const templateStandard = {
  palette: {
    primary: blueGrey,
    secondary: yellow,
    borderColor: '#e6ecf5',
    background: {
      default: '#EDF2F6',
    },
  },
  borderRadius: '5px',
};

const mainTheme = createMuiTheme(templateStandard);

export const gutter = '15px';
export const containerWidth = '1300px';

export default mainTheme;
