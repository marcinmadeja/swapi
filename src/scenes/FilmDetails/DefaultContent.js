import React from 'react';

const DefaultContent = ({ details }) => {
  return (
    <div>
      {details.opening_crawl}
    </div>
  );
};

export default DefaultContent;
