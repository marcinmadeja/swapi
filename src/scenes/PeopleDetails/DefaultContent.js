import React from 'react';
import PropTypes from 'prop-types';

const DefaultContent = ({
  details: {
    height,
    mass,
    hair_color,
    skin_color,
    eye_color,
  },
}) => {
  return (
    <div>
      height: <strong>{height}</strong><br />
      mass: <strong>{mass}</strong><br />
      hair color: <strong>{hair_color}</strong><br />
      skin color: <strong>{skin_color}</strong><br />
      eye color: <strong>{eye_color}</strong><br />
    </div>
  );
};

DefaultContent.propTypes = {
  details: PropTypes.shape({
    height: PropTypes.string.isRequired,
    mass: PropTypes.string.isRequired,
    hair_color: PropTypes.string.isRequired,
    skin_color: PropTypes.string.isRequired,
    eye_color: PropTypes.string.isRequired,
  }).isRequired,
};

export default DefaultContent;
