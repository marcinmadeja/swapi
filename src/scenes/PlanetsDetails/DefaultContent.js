import React from 'react';

const DefaultContent = ({
  details: {
    terrain,
    surface_water,
    population,
    orbital_period,
    rotation_period,
  },
}) => {
  return (
    <div>
      terrain: <strong>{terrain}</strong><br />
      surface water: <strong>{surface_water}</strong><br />
      population: <strong>{population}</strong><br />
      orbital period: <strong>{orbital_period}</strong><br />
      rotation period: <strong>{rotation_period}</strong><br />
    </div>
  );
};

export default DefaultContent;
