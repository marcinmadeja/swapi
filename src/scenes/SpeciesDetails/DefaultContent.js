import React from 'react';

const DefaultContent = ({
  details: {
    average_height,
    skin_colors,
    hair_colors,
    rotation_period,
    eye_colors,
    average_lifespan,
  },
}) => {
  return (
    <div>
      average height: <strong>{average_height}</strong><br />
      skin colors: <strong>{skin_colors}</strong><br />
      hair colors: <strong>{hair_colors}</strong><br />
      eye colors: <strong>{eye_colors}</strong><br />
      average lifespan: <strong>{average_lifespan}</strong><br />
    </div>
  );
};

export default DefaultContent;
