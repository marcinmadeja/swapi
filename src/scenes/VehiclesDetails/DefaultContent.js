import React from 'react';

const DefaultContent = ({
  details: {
    cost_in_credits,
    length,
    crew,
    passengers,
    starship_class,
  },
}) => {
  return (
    <div>
      starship class: <strong>{starship_class}</strong><br />
      crew: <strong>{crew}</strong><br />
      passengers: <strong>{passengers}</strong><br />
      cost in credits: <strong>{cost_in_credits}</strong><br />
      length: <strong>{length}</strong><br />
    </div>
  );
};

export default DefaultContent;
