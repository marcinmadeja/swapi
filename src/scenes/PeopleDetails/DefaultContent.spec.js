import React from 'react';
import 'jest-styled-components';
import { swapiPeople } from 'constants/testConstants';
import DefaultContent from './DefaultContent';

describe('DefaultContent', () => {
  it('should render correctly with data', () => {
    const component = <DefaultContent details={swapiPeople[0]} />;
    const tree = renderer
      .create(component)
      .toJSON();

    expect(tree).toMatchSnapshot(tree);
  });
});
