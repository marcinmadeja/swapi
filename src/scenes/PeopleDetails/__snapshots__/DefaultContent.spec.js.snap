// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[object Object] 1`] = `
<div>
  height: 
  <strong>
    150
  </strong>
  <br />
  mass: 
  <strong>
    49
  </strong>
  <br />
  hair color: 
  <strong>
    brown
  </strong>
  <br />
  skin color: 
  <strong>
    light
  </strong>
  <br />
  eye color: 
  <strong>
    brown
  </strong>
  <br />
</div>
`;
